src/clients/HubPoolClient.ts

```diff
@@ -224,12 +228,12 @@ export class HubPoolClient extends BaseAbstractClient {
    * @param deposit Deposit event
    * @param returns string L1 token counterpart for Deposit
    */
-  getL1TokenForDeposit(deposit: Pick<DepositWithBlock, "quoteBlockNumber" | "originToken" | "originChainId">): string {
+  getL1TokenForDeposit(deposit: Pick<v3DepositWithBlock, "originChainId" | "inputToken" | "quoteBlockNumber">): string {
```

Comment:
How come we don't need a `getL1TokenForV2Deposit` AND a `getL1TokenForV3Deposit`?

Reply:
I thought I could get away with just normalising everything to v3 types (as was done in the SpokePoolClient), but this blew up in relayer-v2. I've subsequently updated this to accept both types. It's pretty ugly because it's a union of `Pick<...>` statements, and ti has to be exported to relayer-v2 since we override this method there as well. However, it's only going to be short-lived so I'm OK with it.

Reply:
Resolved here: https://github.com/across-protocol/sdk-v2/pull/501/commits/8884af524de2eac197de32348b7696e55e5e39a4

I was able to simplify quite a bit by taking advantage of some of the new utility functions in V3Utils.